prow/github/client.go

```diff
@@ -797,6 +797,48 @@ func (c *Client) GetRepos(org string, isUser bool) ([]Repo, error) {
 	return repos, nil
 }
 
+func (c *Client) GetBranches(org, repo string) ([]Branch, error) {
+	c.log("GetBranches", org, repo)
+	var branches []Branch
+	_, err := c.request(&request{
+		method:    http.MethodGet,
+		path:      fmt.Sprintf("%s/repos/%s/%s/branches", c.base, org, repo),
+		exitCodes: []int{200},
+	}, &branches)
+	return branches, err
+}
+
+func (c *Client) RemoveBranchProtection(org, repo, branch string) error {
+	c.log("RemoveBranchProtection", org, repo, branch)
+	_, err := c.request(&request{
+		method:    http.MethodDelete,
+		path:      fmt.Sprintf("%s/repos/%s/%s/branches/%s/protection", c.base, org, repo, branch),
+		exitCodes: []int{204},
+	}, nil)
+	return err
+}
+
+func (c *Client) UpdateBranchProtection(org, repo, branch string, requiredContexts []string, pushers []string) error {
+	c.log("UpdateBranchProtection", org, repo, branch, requiredContexts, pushers)
+	_, err := c.request(&request{
+		method: http.MethodPut,
+		path:   fmt.Sprintf("%s/repos/%s/%s/branches/%s/protection", c.base, org, repo, branch),
+		requestBody: BranchProtectionRequest{
+			RequiredStatusChecks: RequiredStatusChecks{
+				Strict:   false,
+				Contexts: requiredContexts,
+			},
+			EnforceAdmins:              false,
+			RequiredPullRequestReviews: nil,
```

Comment:
I know some repos (like k/kubectl) use pull request reviews as a merge requirement. Has nil been tested to not override this setting? Should we just expose it instead? I have had some issues with Github before not liking nil/non-explicit answers.

Reply:
Let me deal with this in a follow-up PR.

I'm trying to write the minimum set of features, expanding as we need (rather than wasting a bunch of time implementing functionality no one ever uses).

`nil` is explicitly what github wants to disable the functionality https://developer.github.com/v3/repos/branches/#update-branch-protection

My thought is that we may not need this? It seems to duplicate functionality of the approval-plugin.

Reply:
https://github.com/kubernetes/test-infra/issues/6182